### DISCUSSION TITLE
quickFilters : Change account/profile in Move To actions

### DISCUSSION THREAD

**Discussion by TonyGravagno** [RealRaven2000/FiltaQuilla#283] [Ideas]
Consider this optional enhancement for the scenario when you are doing a copy/paste or clone of a filter from one account to another:

If the destination profile is not the same as current, set Move To actions to the destination. Otherwise, unless manually changed, mail in profile2 (where new filter is saved) will get moved to profile1 where the original filter was created.

This might be optional for each move operation, as sometimes we do want all mail in a category to go to a specific profile/folder, regardless of which account received it.

**Comment by RealRaven2000** [comment]
Maybe add as feature request to quickFilters, but needs to be discussed in  detail:

> If the destination profile is not the same as current, set Move To actions to the destination. 

this won't be possible because we cannot assume that the target profile is in any way similar to the source profile. If it is a complete copy of the folder structure I would see this as an exception to the rule and I wouldn't really like quickFilters to start guessing. All folder URIs are absolute paths based on the profile, no  relative paths.

> Otherwise, unless manually changed, mail in profile2 (where new filter is saved) will get moved to profile1 where the original filter was created.

Yes but that may be exactly the point of a clone: gathering all emails that satisfy a condition in a single target folder (and not creating identical copies of mail servers)

**Reply by TonyGravagno** [reply]
> > Otherwise, unless manually changed, mail in profile2 (where new filter is saved) will get moved to profile1 where the original filter was created.

> Yes but that may be exactly the point of a clone: gathering all emails that satisfy a condition in a single target folder (and not creating identical copies of mail servers)

Yup:

> This might be optional for each move operation, as sometimes we do want all mail in a category to go to a specific profile/folder, regardless of which account received it.

**Reply by TonyGravagno** [reply]
> this won't be possible because we cannot assume that the target profile is in any way similar to the source profile. If it is a complete copy of the folder structure I would see this as an exception to the rule

That's the exact scenario that prompted this idea. I'm duplicating structures for two accounts, and I have an equally consistent archive for them. In fact, I'm creating folder structures to accommodate diverse email from common software and I think it would be neat to be able to share these filters with other Tb users. These are opinionated structures of course, but suggestions to help bring order from chaos. That is : If you're having the same problems that I have in organizing mail from these sources, try my filters and structures.

I have other accounts that have identical structures for some filters, not others - so sometimes the target profile is similar and othertimes not.

In another discussion here I propose the ability to edit details on cloned filters, which would give us the ability to change folder names, extending beyond the limits of this feature.

> and I wouldn't really like quickFilters to start guessing.

Agreed.

If a folder doesn't exist in another profile, the exact same condition would exist as if we had done a restore of a quickFilters backup after changing the folder structure. So there is already precedent for this process.

A filter that attempts an action on a non-existent folder will simply not work and will generate a warning. So there's no damage done with this. If the destination is structured differently, the user must simply manually correct the filter, which is what we need to do right now in the above restore scenario.

As an aside, all of these concerns would be negated if Tb had a filter action to create folders that do not exist before filing. That's how the Dovecot Pidgeonhole Sieve implementation works with `fileinto :create "folder.path"`.

> All folder URIs are absolute paths based on the profile, no relative paths.

I'm not proposing relative paths though that would be cool too. 😉

**Reply by RealRaven2000** [reply]
> > All folder URIs are absolute paths based on the profile, no relative paths.
> 
> I'm not proposing relative paths though that would be cool too. 😉

absolute includes the root folder (the server / mailbox)  - here is a typical folder URI:

`      "uri": "imap://axel.grude%40gmail.com@imap.gmail.com/[Google Mail]/Sent Mail"`
